src/Compilers/CSharp/Test/Semantic/Semantics/ValueTupleTests.cs

```diff
@@ -214,5 +215,391 @@ public void TestMissingWellKnownMembersForValueTuple()
+        }
+
+        [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/60961")]
+        public void ExplicitInterfaceImplementation_GenericProperty()
+        {
+            var source = """
+                namespace System
+                {
+                    public struct ValueTuple<T1, T2, T3, T4, T5, T6, T7, TRest> :
+                        System.IEquatable<(T1, T2, T3, T4, T5, T6, T7, TRest)>,
+                        System.Runtime.CompilerServices.ITuple<ValueTuple<T1, T2, T3, T4, T5, T6, T7, TRest>>
+                        where TRest : struct
+                    {
+                        public T1 Item1;
+                        public T2 Item2;
+                        public T3 Item3;
+                        public T4 Item4;
+                        public T5 Item5;
+                        public T6 Item6;
+                        public T7 Item7;
+                        public TRest Rest;
+                        int System.Runtime.CompilerServices.ITuple<ValueTuple<T1, T2, T3, T4, T5, T6, T7, TRest>>.Length => throw null;
+                        ValueTuple<T1, T2, T3, T4, T5, T6, T7, TRest> System.Runtime.CompilerServices.ITuple<ValueTuple<T1, T2, T3, T4, T5, T6, T7, TRest>>.Self => throw null;
+                        object System.Runtime.CompilerServices.ITuple<ValueTuple<T1, T2, T3, T4, T5, T6, T7, TRest>>.this[int index] => throw null;
+                        ValueTuple<T1, T2, T3, T4, T5, T6, T7, TRest> System.Runtime.CompilerServices.ITuple<ValueTuple<T1, T2, T3, T4, T5, T6, T7, TRest>>.this[ValueTuple<T1, T2, T3, T4, T5, T6, T7, TRest> self, int index] => throw null;
+                        bool System.IEquatable<(T1, T2, T3, T4, T5, T6, T7, TRest)>.Equals((T1, T2, T3, T4, T5, T6, T7, TRest) other) => false;
+                    }
+
+                    public interface IEquatable<T>
+                    {
+                        bool Equals(T other);
+                    }
+
+                    public class Object { }
+                    public class ValueType { }
+                    public struct Void { }
+                    public struct Boolean { }
+                    public struct Int32 { }
+                    public struct IntPtr { }
+                    public class Exception { }
+                    public class String { }
+                    public struct ValueTuple<T> { }
+
+                    namespace Runtime.CompilerServices
+                    {
+                        public interface ITuple<T> where T : ITuple<T>
+                        {
+                            int Length { get; }
+                            T Self { get; }
+                            object this[int index] { get; }
+                            T this[T self, int index] { get; }
+                        }
+                    }
+                }
+                """;
+
+            var comp = CreateEmptyCompilation(source);
+            comp.VerifyDiagnostics();
+        }
```

Comment:
> } [](http://example.com/codeflow?start=7&length=2)

It looks like there are no tests with tuple types used in the signature of the implemented member (I recommended to add them in our offline conversation)

Reply:
I thought I added them. For example, `ExplicitInterfaceImplementation_Indexer` has `bool System.IEquatable<(T1, T2, T3, T4, T5, T6, T7, TRest)>.Equals((T1, T2, T3, T4, T5, T6, T7, TRest) other) => false;` - the implemented member has tuple types in signature. Or did you mean like non-generic tuple types?

Reply:
> . Or did you mean like non-generic tuple types?

I meant tests that do not refer to tuples anywhere, but in the signature of the implemented memeber.